src/Agent.Listener/DistributedTask.Pipelines/Yaml/PipelineParser.cs

```diff
@@ -299,21 +300,21 @@ private void ResolveTemplates(Process process, String defaultRoot, CancellationT
             {
                 ResolveTemplates(process.Phases, defaultRoot, cancellationToken, ref fileCount);
             }
-            else if (process.Jobs != null)
-            {
-                ResolveTemplates(process.Jobs, defaultRoot, cancellationToken, ref fileCount);
-            }
-            else if (process.Variables != null)
+            else
             {
-                ResolveTemplates(process.Variables, defaultRoot, cancellationToken, ref fileCount);
-            }
-            else if (process.Steps != null)
-            {
-                ResolveTemplates(process.Steps, defaultRoot, cancellationToken, ref fileCount);
+                if (process.Variables != null)
```

Comment:
`phases` cannot be defined in the root of the yaml file when `variables` or `steps` is also defined (`phases` is process-level; `variables` and `steps` are phase-level).

It looks like I had an artificial limitation in the way this was previously implemented (resolving templates in variables and steps were mutally exclusive code paths).